meta-balena-common/recipes-containers/docker-disk/files/balena-api.inc

```diff
@@ -0,0 +1,283 @@
+#!/bin/bash
+
+CURL="curl -s"
+
+TRANSLATION=${TRANSLATION:-"v6"}
+
+DEBUG=0
+pp_json() {
+	if [ "${DEBUG}" = "1" ]; then
+		echo "${1}" | >&2 jq .
+	fi
+}
+
+dlog() {
+	if [ "${DEBUG}" = "1" ]; then
+		>&2 echo "${1}"
+	fi
+}
+
+# Returns success if the check fails
+check_fail() {
+	local _json
+	local _msg
+	_json="$1"
+	_msg="$2"
+
+	if [ "${_json}" != "OK" ]; then
+		pp_json "${_json}"
+		>&2 echo "${_msg}"
+		return 0
+	fi
+	return 1
+}
+
+# Print public application ID from application name
+# Arguments:
+#
+# $1: Application name
+# $2: Balena target environment
+#
+# Result:
+# Prints the application ID or null if it does not exist
+print_appID_from_appName() {
+	local _appName="$1"
+	local _apiEnv="$2"
+	[ -z "${_appName}" ] && >&2 echo "Application name is required" && return 1
+	local _appID=""
+	local _json=""
+	_json=$(${CURL} -XGET "https://api.${_apiEnv}/${TRANSLATION}/application?\$filter=(app_name%20eq%20'${_appName}')%20and%20(is_public%20eq%20true)" -H "Content-Type: application/json")
+	pp_json "${_json}"
+	_appID=$(echo "${_json}" | jq --raw-output '.d[0].id')
+	dlog "[${_appName}] Application ID is ${_appID}"
+	echo "${_appID}"
+}
+
+# Print UUID from public application name
+# Arguments:
+#
+# $1: Application name
+# $2: Balena target environment
+#
+# Result:
+# Prints the application UUID or nothing if it does not exist
+print_appUUID_from_appName() {
+	local _appName="$1"
+	local _apiEnv="$2"
+	[ -z "${_appName}" ] && >&2 echo "Application name is required" && return 1
+	local _appID=""
+	local _json=""
+	_json=$(${CURL} -XGET "https://api.${_apiEnv}/resin/application?\$filter=(app_name%20eq%20'${_appName}')%20and%20(is_public%20eq%20true)" -H "Content-Type: application/json")
```

Comment:
```suggestion
	_json=$(${CURL} -XGET "https://api.${_apiEnv}/${TRANSLATION}/application?\$filter=(app_name%20eq%20'${_appName}')%20and%20(is_public%20eq%20true)" -H "Content-Type: application/json")
```

Reply:
two more things here:

1. it might make more sense to use the full application `slug`, assuming that hostapps can only come from us/the `balena_os` org to start (similar to https://github.com/balena-os/meta-balena/pull/2052)
1. if this is an unprivileged call (which it seems to be since there are no auth headers), you can drop the `is_public eq true` filter as only public apps are returned for unauthorized queries.

Reply:
Done.

Reply:
Also, the UUID is currently present only in the root endpoing "https://api.${_apiEnv}/resin/application" so there is no translation for this field.

Reply:
@alexgg is this code in the long tail, or is it only called at build time? if it is in the long tail, i think we absolutely cannot rely on the `resin` endpoint as that changes freely. we can either a) backport the uuid change to `v6` or b) perhaps cut `v7`? cc @Page- for guidance.

Reply:
Additionally i missed the earlier question, but the `balena_os` namespace is reserved at API startup time. Even in open-balena i think the plan is to allow users to sync with our upstream apps for things like the OS / supervisor.

Reply:
thanks @xginn8, I will change to use the full slug as you suggested. About the UUID translation, it's only used at build time when creating the `apps.json` with the preloaded extra containers. I will use a translation once there is one, but I'd prefer not to hold the release of host extensions on yet another variable if possible.

Reply:
I think adding it to v6 is how we'd go @xginn8, we tend to only do new versions for breaking changes. And absolutely nothing in the os should ever rely on `/resin/` @alexgg because it will break

Reply:
Created https://github.com/balena-io/balena-api/issues/2927 to do the above